src/process/personality.rs

```diff
@@ -0,0 +1,1695 @@
+    };
+
+    prctl_3args(PR_MCE_KILL, sub_operation as *mut _, policy).map(|_r| ())
+}
+
+/*
+ * PR_SET_MM
+ */
+
+const PR_SET_MM: c_int = 35;
+
+const PR_SET_MM_START_CODE: u32 = 1;
+const PR_SET_MM_END_CODE: u32 = 2;
+const PR_SET_MM_START_DATA: u32 = 3;
+const PR_SET_MM_END_DATA: u32 = 4;
+const PR_SET_MM_START_STACK: u32 = 5;
+const PR_SET_MM_START_BRK: u32 = 6;
+const PR_SET_MM_BRK: u32 = 7;
+const PR_SET_MM_ARG_START: u32 = 8;
+const PR_SET_MM_ARG_END: u32 = 9;
+const PR_SET_MM_ENV_START: u32 = 10;
+const PR_SET_MM_ENV_END: u32 = 11;
+const PR_SET_MM_AUXV: usize = 12;
+const PR_SET_MM_EXE_FILE: usize = 13;
+const PR_SET_MM_MAP: usize = 14;
+const PR_SET_MM_MAP_SIZE: usize = 15;
+
+/// `PR_SET_MM_*`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(u32)]
+pub enum ProcessVirtualMemoryMapAddress {
+    /// Set the address above which the program text can run.
+    CodeStart = PR_SET_MM_START_CODE,
+    /// Set the address below which the program text can run.
+    CodeEnd = PR_SET_MM_END_CODE,
+    /// Set the address above which initialized and uninitialized (bss) data are placed.
+    DataStart = PR_SET_MM_START_DATA,
+    /// Set the address below which initialized and uninitialized (bss) data are placed.
+    DataEnd = PR_SET_MM_END_DATA,
+    /// Set the start address of the stack.
+    StackStart = PR_SET_MM_START_STACK,
+    /// Set the address above which the program heap can be expanded with `brk()` call.
+    BrkStart = PR_SET_MM_START_BRK,
+    /// Set the current `brk()` value.
+    BrkCurrent = PR_SET_MM_BRK,
+    /// Set the address above which the program command line is placed.
+    ArgStart = PR_SET_MM_ARG_START,
+    /// Set the address below which the program command line is placed.
+    ArgEnd = PR_SET_MM_ARG_END,
+    /// Set the address above which the program environment is placed.
+    EnvironmentStart = PR_SET_MM_ENV_START,
+    /// Set the address below which the program environment is placed.
+    EnvironmentEnd = PR_SET_MM_ENV_END,
+}
+
+/// Modify certain kernel memory map descriptor addresses of the calling process.
+#[inline]
+pub fn set_process_virtual_memory_map_address(
+    option: ProcessVirtualMemoryMapAddress,
+    address: *const u8,
+) -> io::Result<()> {
+    prctl_3args(PR_SET_MM, option as usize as *mut _, address as *mut _).map(|_r| ())
+}
+
+/// Supersede the `/proc/pid/exe` symbolic link with a new one pointing to a new executable file.
+#[inline]
+pub fn set_process_executable_file(fd: BorrowedFd) -> io::Result<()> {
+    let fd = usize::try_from(fd.as_raw_fd()).map_err(|_r| io::Errno::RANGE)?;
+    prctl_3args(PR_SET_MM, PR_SET_MM_EXE_FILE as *mut _, fd as *mut _).map(|_r| ())
+}
+
+/// Set a new auxiliary vector.
+#[inline]
+pub fn set_process_auxiliary_vector(auxv: &[u64]) -> io::Result<()> {
+    syscalls::prctl(
+        PR_SET_MM,
+        PR_SET_MM_AUXV as *mut _,
+        auxv.as_ptr() as *mut _,
+        auxv.len() as *mut _,
+        ptr::null_mut(),
+    )
+    .map(|_r| ())
+}
```

Comment:
> This should be an unsafe function, because libc could later look up the aux information and interpret the values as pointers that it can dereference.

[This lint](https://github.com/bytecodealliance/rustix/blob/d96706a9b3d7126161af595c9c3fa40afbd1dce3/src/lib.rs#L98) forbids me from that.

> Also, instead of u64, aux vectors use pointer-sized elements, and some of them are pointers, so can this use *mut c_void instead, to preserve provenance?

Sure.